rebuild/src/secret_key.cc

```diff
@@ -0,0 +1,218 @@
+#include "secret_key.h"
+
+void SecretKey::Init(Napi::Env env, Napi::Object &exports, BlstTsAddon *module)
+{
+    Napi::HandleScope scope(env); // no need to EscapeHandleScope, Persistent will take care of it
+    auto proto = {
+        StaticMethod("fromKeygen", &SecretKey::FromKeygen, static_cast<napi_property_attributes>(napi_static | napi_enumerable)),
+        StaticMethod("deserialize", &SecretKey::Deserialize, static_cast<napi_property_attributes>(napi_static | napi_enumerable)),
+        InstanceMethod("serialize", &SecretKey::Serialize, static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod("toPublicKey", &SecretKey::ToPublicKey, static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod("sign", &SecretKey::Sign, static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "SecretKey", proto, module);
+    module->_secret_key_ctr = Napi::Persistent(ctr);
+    module->_secret_key_tag = {BLST_TS_SECRET_KEY_LOWER_TAG, BLST_TS_SECRET_KEY_UPPER_TAG};
+    exports.Set(Napi::String::New(env, "SecretKey"), ctr);
+}
+
+Napi::Value SecretKey::FromKeygen(const Napi::CallbackInfo &info)
+{
+    Napi::Env env = info.Env();
+    Napi::EscapableHandleScope scope(env);
+
+    // Pull Value from info and check type is valid
+    Napi::Value ikm_value = info[0];
```

Comment:
may be we can define  a macro to check for typed array and reduce redundant code

Reply:
Updated. I also created one for unwrapped objects.